xarray/core/groupers.py

```diff
@@ -41,15 +41,20 @@
 class EncodedGroups:
     """
     Dataclass for storing intermediate values for GroupBy operation.
-    Returned by factorize method on Grouper objects.
+    Returned by the ``factorize`` method on Grouper objects.
 
-    Parameters
+    Attributes
     ----------
-    codes: integer codes for each group
-    full_index: pandas Index for the group coordinate
-    group_indices: optional, List of indices of array elements belonging
-                   to each group. Inferred if not provided.
-    unique_coord: Unique group values present in dataset. Inferred if not provided
+    codes: DataArray
+        Same shape as the DataArray to group by. Values consist of a unique integer code for each group.
+    full_index: pd.Index
+        Pandas Index for the group coordinate containing unique group labels.
+        This can differ from ``unique_coord`` in the case of resampling and binning,
+        where certain groups in the output need not be present in the input.
+    group_indices: list of int or slice or list of int, optional
```

Comment:
Do we need `group_indices` to be mutable? Should it be `tuple[int, ...]` instead?